src/metabase/driver.clj

```diff
@@ -42,6 +44,42 @@
    :username-incorrect                 "Looks like your username is incorrect."
    :username-or-password-incorrect     "Looks like the username or password is incorrect."})
 
+(def AnalyzeTable
+  "Schema for the expected output of `analyze-table`."
+  {(schema/optional-key :row_count) schema/Int
+   (schema/optional-key :fields)    [{:id                                    schema/Int
+                                      (schema/optional-key :special-type)    (apply schema/enum field/special-types)
+                                      (schema/optional-key :preview-display) schema/Bool
+                                      (schema/optional-key :values)          [schema/Any]}]})
+
+(def DescribeDatabase
+  "Schema for the expected output of `describe-database`."
+  {:tables #{{:name                         schema/Str
+              (schema/optional-key :schema) (schema/maybe schema/Str)}}})
+
+(def DescribeTableField
+  "Schema for a given Field as provided in `describe-table` or `analyze-table`."
```

Comment:
There are small differences right now, but we could certainly make some tweaks and end up with a more unified schema for a TableField.

Right now AnalyzeTable expects the `:id` of each field to be returned because it makes it easier to align the data back to the fields we've already synced.  We could skip that if we wanted to and match based on name i suppose.

Also we only allow setting `:values` and `:preview-display` when analyzing a table, but that could be changed as well.

The current schema is accurate, so I'd recommend just going ahead and merging this.  Modifying the code to prefer a different schema is really a different task, but I agree it would make sense to end up with a single expected schema for a TableField that can be reused across both functions.